pkg/analysis/nomaps/testdata/src/a/a.go

```diff
@@ -0,0 +1,34 @@
+package a
+
+type (
+	DefinedType  map[string]Component
+	DefinedType2 *map[string]Component
+	DefinedType3 map[string]int
```

Comment:
What about types that aren't `map[string]T`? What about `map[int]T` for example? Would we catch that today?

Reply:
Added generics cases. And all are caught.